site/content/contribute/mobile/e2e/ios.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "Setup and Run iOS E2E"
+date: 2020-09-01T09:00:00-00:00
+weight: 4
+subsection: Mobile End-to-End Testing
+---
+
+#### Local setup
+
+1. Install [applesimutils](https://github.com/wix/AppleSimulatorUtils):
+```
+brew tap wix/brew
+brew install applesimutils
+```
+2. Set XCode's build location so that the built app, especially debug, is expected at the project's location instead of the Library's folder which is unique/hashed.
+3. Open XCode, then go to **XCode > Preferences > Locations**.
+4. Under **Derived Data**, click **Advanced...**.
+5. Select **Custom > Relative to Workspace**, then set **Products** as **Build/Products**.
+6. Click **Done** to save the changes.
```

Comment:
Isn't this already defined by the iOS project? can we verify this is actually required? specially  as there is no way to do this on the CI

Reply:
Yes, this is required in local development so we know where to pick up the [binary](https://github.com/mattermost/mattermost-mobile/blob/master/detox/.detoxrc.json#L6). If not, the binary will be placed in the default XCode folder with non-deterministic and dynamically generated hash folder. No change in CI - still on [`/build-ios` folder](https://github.com/mattermost/mattermost-mobile/blob/master/fastlane/Fastfile#L362-L365).